net-vpn/i2pd/i2pd-2.28.0.ebuild

```diff
@@ -0,0 +1,105 @@
+# Copyright 1999-2019 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+inherit systemd cmake-utils toolchain-funcs
+
+DESCRIPTION="A C++ daemon for accessing the I2P anonymous network"
+HOMEPAGE="https://github.com/PurpleI2P/i2pd"
+SRC_URI="https://github.com/PurpleI2P/${PN}/archive/${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="BSD"
+SLOT="0"
+KEYWORDS="~amd64 ~arm ~arm64 ~ia64 ~ppc ~ppc64 ~x86"
+IUSE="cpu_flags_x86_aes cpu_flags_x86_avx i2p-hardening libressl static +upnp websocket"
+
+# if using libressl, require >=boost-1.65, see #597798
+RDEPEND="
+	acct-user/i2pd
+	acct-group/i2pd
+	!static? (
+		dev-libs/boost:=[threads]
+		!libressl? ( dev-libs/openssl:0=[-bindist] )
+		libressl? (
+			dev-libs/libressl:0=
+			>=dev-libs/boost-1.65:=
+		)
+		upnp? ( net-libs/miniupnpc )
+	)"
+DEPEND="${RDEPEND}
+	static? (
+		dev-libs/boost:=[static-libs,threads]
+		!libressl? ( dev-libs/openssl:0=[static-libs] )
+		libressl? (
+			dev-libs/libressl:0=[static-libs]
+			>=dev-libs/boost-1.65:=
+		)
+		sys-libs/zlib:=[static-libs]
+		upnp? ( net-libs/miniupnpc[static-libs] )
+	)
+	websocket? ( dev-cpp/websocketpp )"
+
+CMAKE_USE_DIR="${S}/build"
+
+DOCS=( README.md contrib/i2pd.conf contrib/tunnels.conf )
+
+PATCHES=( "${FILESDIR}/${PN}-2.14.0-fix_installed_components.patch"
+	"${FILESDIR}/i2pd-2.25.0-lib-path.patch" )
+
+pkg_pretend() {
+	if tc-is-gcc && ! ver_test "$(gcc-version)" -ge "4.7"; then
+		die "At least gcc 4.7 is required"
+	fi
```

Comment:
Is this _really_ necessary? The profile defaults are way newer than 4.7...

Reply:
I am not sure. gcc-3.4 is still available and stable in the tree.

Reply:
Yes but all the profiles default to newer ones. I dont even know what one needs to do to be able to use 4.7 with current toolchain. 

Btw I merged tihs already so please dont commit anything on this PR anymore :P

Reply:
ok, thanks!! I may remove the compiler check on next bump. It is exotic indeed.